core/src/main/java/org/apache/iceberg/hadoop/HadoopTables.java

```diff
@@ -144,6 +147,52 @@ public Table create(Schema schema, PartitionSpec spec, SortOrder order,
     return new BaseTable(ops, location);
   }
 
+  /**
+   * Drop a table and delete all data and metadata files.
+   *
+   * @param location a path URI (e.g. hdfs:///warehouse/my_table)
+   * @return true if the table was dropped
+   * @throws NoSuchTableException if the table does not exists.
+   */
+  public boolean dropTable(String location) {
+    return dropTable(location, true);
+  }
+
+  /**
+   * Drop a table; optionally delete data and metadata files.
+   * <p>
+   * If purge is set to true the implementation should delete all data and metadata files.
+   *
+   * @param location a path URI (e.g. hdfs:///warehouse/my_table)
+   * @param purge if true, delete all data and metadata files in the table
+   * @return true if the table was dropped
+   * @throws NoSuchTableException if the table does not exists.
+   */
+  public boolean dropTable(String location, boolean purge) {
+    TableOperations ops = newTableOps(location);
+    TableMetadata lastMetadata = null;
+    if (ops.current() != null) {
+      if (purge) {
+        lastMetadata = ops.current();
+      }
+    } else {
+      throw new NoSuchTableException("Table does not exist at location: %s, so it can not be dropped", location);
```

Comment:
I should have caught this yesterday, but shouldn't this return `false` instead of throwing the exception? That's what all the other `drop` methods do. If the table doesn't exist, it isn't an exceptional case. It just returns false to signal that it nothing needed to be done.

Reply:
> I should have caught this yesterday, but shouldn't this return `false` instead of throwing the exception? That's what all the other `drop` methods do. If the table doesn't exist, it isn't an exceptional case. It just returns false to signal that it nothing needed to be done.

Good point! Finding it now is better than finding it after pushing the PR 😄 
Fixed.